tests/test_dataset.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import pytest as pytest
+
+from sparsezoo.utils import Dataset
+
+
+@pytest.mark.parametrize(
+    "_data",
+    [
+        Dataset(data=np.empty(shape=(100, 100, 10, 10)), name="4d"),
+    ],
+)
+def test_has_iter_batches(_data):
+    assert hasattr(_data, "iter_batches")
+
+
+@pytest.mark.parametrize(
+    "_data",
+    [
+        Dataset(data=np.empty(shape=(100, 100, 10, 10)), name="4d"),
```

Comment:
input to dataset should be an iterable of numpy list(s), instead of a single one